pkg/cloudprovider/aws/apis/v1alpha1/provider.go

```diff
@@ -18,10 +18,19 @@ import (
 	"encoding/json"
 	"fmt"
 
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
 	"github.com/aws/karpenter/pkg/apis/provisioning/v1alpha5"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const (
+	DefaultMetadataOptionsHTTPEndpoint            = ec2.LaunchTemplateInstanceMetadataEndpointStateEnabled
+	DefaultMetadataOptionsHTTPProtocolIPv6        = ec2.LaunchTemplateInstanceMetadataProtocolIpv6Disabled
+	DefaultMetadataOptionsHTTPPutResponseHopLimit = 2
+	DefaultMetadataOptionsHTTPTokens              = ec2.LaunchTemplateHttpTokensStateRequired
```

Comment:
I think we should default to Optional. Having IMDSv2 be required is the right thing to be doing long-term but I don't think every application in the wild is there yet and forcing them to always specify metadata in the providerSpec might be too heavy-handed

Reply:
Alright let's wait for @bwagner5  to ping here before you change it again :)

Reply:
Sounds good, let's make sure to call that out in the release notes since we've got the launch templates are currently configured with v1v2. Thanks for chiming in!